go/cli/mcap/cmd/merge.go

```diff
@@ -82,6 +93,30 @@ func (m *mcapMerger) outputSchemaID(inputID int, inputSchemaID uint16) (uint16,
 	return v, ok
 }
 
+func getChannelHash(channel *mcap.Channel, coalesceChannels string) string {
+	hasher := md5.New()
+	schemaIDBytes := make([]byte, 2)
+	binary.LittleEndian.PutUint16(schemaIDBytes, channel.SchemaID)
+	hasher.Write(schemaIDBytes)
+	hasher.Write([]byte(channel.Topic))
+	hasher.Write([]byte(channel.MessageEncoding))
+
+	switch coalesceChannels {
+	case AutoCoalescing: // Include channel metadata in hash
+		for key, value := range channel.Metadata {
+			hasher.Write([]byte(key))
+			hasher.Write([]byte(value))
+		}
+	case ForceCoalescing: // Channel metadata is not included in hash
+		break
+	default:
+		die("Invalid value for --coalesce-channels: %s\n", coalesceChannels)
+	}
+
+	hash := hasher.Sum(nil)
+	return hex.EncodeToString(hash)
```

Comment:
seems odd to me that we store the hash as a string rather than a fixed size value - won't block the PR, just seems odd.

Reply:
I feel like channelIdByHash should key off the result of `hasher.Sum()`.

Reply:
Done. Should I do the same for the `schemaIDByHash` map ?

Reply:
Yep, that would be good.